src/mower_logic/src/mower_logic/mower_logic.cpp

```diff
@@ -344,6 +344,14 @@ void setEmergencyMode(bool emergency)
 }
 
 void updateUI(const ros::TimerEvent &timer_event) {
+    
+    if(currentBehavior == &MowingBehavior::INSTANCE) {
+        mowingBehaviorDowncast = dynamic_cast < MowingBehavior > ( currentBehavior );
```

Comment:
Would cast to a MowingBehavior*.

Need to check for null too, something like:
```
  if (MowingBehavior* mb = dynamic_cast<MowingBehavior*>(currentBehavior); mb != nullptr) {
    high_level_status.current_area = mb->get_current_area();
  }
```

Reply:
actually don't need it I expect, can just call MowingBehaviour::INSTANCE.get_current_area()

Reply:
ah ok. It compiled with your previous suggestion but I'll try now with the above.

Reply:
hmm, havent managed to get it working the last way,

Reply:
`MowingBehaviour::INSTANCE.get_current_area()`. No & (that gets a pointer to it)

Reply:
Not creating an instance, that is the instance (it's a singleton)

Reply:
i believe I already tried that first but it said that MowingBehavior hadn't been declared. Almost identical message less the '&'

Reply:
ahhhh. i might have typed it wrong!  US-UK English... behavior vs behaviour.... let me retry!

Reply:
or you did! or one of us did 😂

Reply:
ah, get's me all the time!